drive/activity-v2/index.html

```diff
@@ -0,0 +1,189 @@
+       *  listeners.
+       */
+      function initClient() {
+        gapi.client.init({
+          apiKey: API_KEY,
+          clientId: CLIENT_ID,
+          discoveryDocs: DISCOVERY_DOCS,
+          scope: SCOPES
+        }).then(function () {
+          // Listen for sign-in state changes.
+          gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+          // Handle the initial sign-in state.
+          updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+          authorizeButton.onclick = handleAuthClick;
+          signoutButton.onclick = handleSignoutClick;
+        }, function(error) {
+          appendPre(JSON.stringify(error, null, 2));
+        });
+      }
+
+      /**
+       *  Called when the signed in status changes, to update the UI
+       *  appropriately. After a sign-in, the API is called.
+       */
+      function updateSigninStatus(isSignedIn) {
+        if (isSignedIn) {
+          authorizeButton.style.display = 'none';
+          signoutButton.style.display = 'block';
+          listDriveActivity();
+        } else {
+          authorizeButton.style.display = 'block';
+          signoutButton.style.display = 'none';
+        }
+      }
+
+      /**
+       *  Sign in the user upon button click.
+       */
+      function handleAuthClick(event) {
+        gapi.auth2.getAuthInstance().signIn();
+      }
+
+      /**
+       *  Sign out the user upon button click.
+       */
+      function handleSignoutClick(event) {
+        gapi.auth2.getAuthInstance().signOut();
+      }
+
+      /**
+       * Append a pre element to the body containing the given message
+       * as its text node. Used to display the results of the API call.
+       *
+       * @param {string} message Text to be placed in pre element.
+       */
+      function appendPre(message) {
+        var pre = document.getElementById('content');
+        var textContent = document.createTextNode(message + '\n');
+        pre.appendChild(textContent);
+      }
+
+      /**
+       * Print recent activity.
+       */
+      function listDriveActivity() {
+        gapi.client.driveactivity.activity
+            .query({
+              'pageSize': 10,
+              'consolidationStrategy': { 'legacy': {} }
+            })
+            .then(function(response) {
+              appendPre('Recent activity:');
+              var activities = response.result.activities;
+
+              if (activities && activities.length > 0) {
+                for (var i = 0; i < activities.length; i++) {
+                  var activity = activities[i];
+                  var time = getTimeInfo(activity);
+                  var action = getActionInfo(activity['primaryActionDetail']);
+                  var actors = activity.actors.map(getActorInfo);
+                  var targets = activity.targets.map(getTargetInfo);
+                  appendPre(
+                      time + ': ' + truncated(actors) + ', ' + action + ', ' +
```

Comment:
Optional: We aren't using it much (if at all) in this repo yet, but you can utilize template literals to make this a little cleaner: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Done.